toolkit/pallets/session-validator-management/src/tests.rs

```diff
@@ -97,18 +101,77 @@ mod inherent_tests {
 		let genesis_validators = [ALICE, BOB];
 		new_test_ext_with_genesis_initial_authorities(&genesis_validators).execute_with(|| {
 			assert!(pallet::NextCommittee::<Test>::get().is_none());
+			let (inherent_data_from_which_committee_cannot_be_selected, selection_inputs_hash) =
+				create_inherent_data(&[]);
 			let call = pallet::Call::set {
 				validators: ids_and_keys_fn(&genesis_validators),
 				for_epoch_number: 43,
+				selection_inputs_hash,
 			};
-			let inherent_data_from_which_committee_cannot_be_selected = create_inherent_data(&[]);
 			// Should be Ok, because check_inherent should use CurrentCommittee value.
 			assert_ok!(<SessionCommitteeManagement as ProvideInherent>::check_inherent(
 				&call,
 				&inherent_data_from_which_committee_cannot_be_selected
 			));
 		});
 	}
+
+	#[test]
+	fn check_inherent_error_includes_both_hashes_if_different() {
```

Comment:
Cosmetic: test name is invalid, because the the error is this:
```
InherentError::InvalidValidatorsMatchingHash(SizedByteString(0xf8317371a8d8b47c9933f038c36fa4587e40635a3648d9f6cd699e4ea7b90ee7))
```